client/components/App.jsx

```diff
@@ -111,82 +46,62 @@ class App extends React.Component {
-
-    const filteredItems = activeFilters.length > 0
-      ? items?.filter((elem) => {
-        const intersection = elem.filterIds.filter((x) => activeFilters.includes(x));
-        return (
-          intersection.length > 0
-              && intersection.length <= elem.filterIds.length
-              && intersection.length === activeFilters.length
-        );
-      })
-      : items;
-
-    return (
-      <div className="app">
-        <nav className="side-nav">
-          <h1>P.</h1>
-          <p>categories</p>
-          <ul>{categoriesList}</ul>
-        </nav>
-        <div className="main">
-          <header>
-            <Filters
-              tags={filters}
-              onSwitch={this.switchFilterCallback}
-              onSwitchAll={this.switchDisplayAllCallback}
-              all={isAllFilters}
-            />
-            <OrderStatus orders={orders} onClick={this.switchOrdersCallback} />
-          </header>
-          <List items={filteredItems} title={categoryTitle} onAdd={App.addOrder} />
-        </div>
-        {isOrdersVisible ? (
-          <Orders orders={orders} discounts={discounts} onClose={this.switchOrdersCallback} />
-        ) : (
-          ''
-        )}
+  };
+
+  const getSwitchFilterCallback = () => (id) => dispatch(switchFilter(id));
+  const getSwitchAllCallback = () => () => dispatch(switchAll());
+  const getSwitchOrdersCallback = () => () => dispatch(switchOrders());
+  const getPostOrdersCallback = () => (id) => dispatch(postOrder(id));
+
+  // render
+  const categoriesLst = categoriesList(data.selectedCategory);
+  const categoryTitle = data.categories.find((elem) => elem._id === data.selectedCategory)?.title;
+  const filters = data.filters.map((elem) => ({
+    isActive: activeFilters.includes(elem._id),
+    ...elem,
+  }));
+
+  const filteredItems = activeFilters.length > 0
+    ? data.items?.filter((elem) => {
+      const intersection = elem.filterIds.filter((x) => activeFilters.includes(x));
+      return (
+        intersection.length > 0
+            && intersection.length <= elem.filterIds.length
+            && intersection.length === activeFilters.length
+      );
+    })
+    : data.items;
+
+  return (
+    <div className="app">
+      <nav className="side-nav">
+        <h1>P.</h1>
+        <p>categories</p>
+        <ul>{categoriesLst}</ul>
+      </nav>
+      <div className="main">
+        <header>
+          <Filters
+            tags={filters}
+            onSwitch={getSwitchFilterCallback()}
+            onSwitchAll={getSwitchAllCallback()}
+            all={isAllFilters}
+          />
+          <OrderStatus orders={data.orders} onClick={getSwitchOrdersCallback()} />
+        </header>
+        <List items={filteredItems} title={categoryTitle} onAdd={getPostOrdersCallback()} />
       </div>
-    );
-  }
+      {isOrdersVisible ? (
+        <Orders
+          orders={data.orders}
+          discounts={data.discounts}
+          onClose={getSwitchOrdersCallback()}
```

Comment:
Try to avoid creating new objects/arrays/functions during render method, for example `getSwitchOrdersCallback()` returns new function, which returns new function, and on each call it returns different `link`, so `<Orders` component will be rerendered, even when nothing changes.